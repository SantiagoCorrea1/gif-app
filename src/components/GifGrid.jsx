import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs"

export const GifGrid = ({ category }) => {

    const { images, isLoding } = useFetchGifs( category );

    return (
        <>
            <h3>{ category }</h3>

            {
                isLoding && ( <h2>cargando...</h2> )
            }

            <div className="card-grid">
                { images.map( ( image ) => (
                    <GifGridItem 
                        key={ image.id }
                        { ...image }
                    />
                ) ) }
            </div>
        </>
    )
}

 