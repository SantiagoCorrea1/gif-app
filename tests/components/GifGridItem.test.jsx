import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components/GifGridItem";

describe('test in GifGridItem', () => {
    
    const title = 'soy un tÃ­tulo';
    const url = 'https://hola.com'

    test('should match the snapShot', () => {

        const { container } = render( <GifGridItem title={ title } url={ url } /> );
        expect( container ).toMatchSnapshot();


    });

    test('should show the image with the URL and ALT given', () => {
        
        render( <GifGridItem title={ title } url={ url } /> );
        // expect ( screen.getByRole( 'img' ).src ).toContain( url );
        // expect ( screen.getByRole( 'img' ).alt ).toContain( title );
        const { src, alt } = screen.getByRole( 'img' );
        expect( src ).toContain( url );
        expect( alt ).toContain( title );

    });

    test('should show the component title', () => {
        render( <GifGridItem title={ title } url={ url } /> );
        expect( screen.getByText( title ) ).toBeTruthy();

    })
    


    

})
