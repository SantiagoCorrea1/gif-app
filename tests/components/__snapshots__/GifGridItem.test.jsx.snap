// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in GifGridItem should match the snapShot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="soy un título"
      src="https://hola.com"
    />
    <p>
      soy un título
    </p>
  </div>
</div>
`;
